packages/application-shell/src/components/application-shell/application-shell.spec.js

```diff
@@ -633,3 +639,596 @@ describe('when dispatching a loading notification', () => {
+  let operations;
+  beforeEach(() => {
+    operations = mocksForMc.createMockOperations({
+      FetchProject: {
+        project: ProjectMock.build({
+          allAppliedPermissions: [{ name: 'canManageOrders', value: true }],
+          allAppliedActionRights: [
+            {
+              group: 'orders',
+              name: 'canAddOrders',
+              value: true,
+            },
+          ],
+        }),
+      },
+    });
+    createGraphqlMockServer(xhrMock, {
+      operationsByTarget: { mc: operations },
+    });
+    selectProjectKeyFromUrl.mockReturnValue(
+      operations.FetchLoggedInUser.me.defaultProjectKey
+    );
+  });
+  it('should render item', async () => {
+    const navbarMock = ApplicationNavbarMenuMock.build({
+      permissions: ['ManageOrders'],
+      actionRights: [{ group: 'orders', name: 'AddOrders' }],
+    });
+    const rendered = renderApp(null, {
+      featureFlags: {
+        [PROJECT_EXTENSIONS]: false,
+      },
+      DEV_ONLY__loadNavbarMenuConfig: () => Promise.resolve([navbarMock]),
+    });
+    // Get the nav container, to narrow down the search area
+    const container = await rendered.findByTestId('left-navigation');
+    const navbarRendered = within(container);
+
+    const applicationLocale = operations.FetchLoggedInUser.me.language;
+    const mainMenuLabel = navbarMock.labelAllLocales.find(
+      localized => localized.locale === applicationLocale
+    );
+    await navbarRendered.findByText(mainMenuLabel.value);
+  });
+});
+describe('when navbar menu items do not match given action rights', () => {
+  let operations;
+  beforeEach(() => {
+    operations = mocksForMc.createMockOperations({
+      FetchProject: {
+        project: ProjectMock.build({
+          allAppliedPermissions: [{ name: 'canManageOrders', value: true }],
+          allAppliedActionRights: [
+            {
+              group: 'orders',
+              name: 'canAddOrders',
+              value: false,
+            },
+          ],
+        }),
+      },
+    });
+    createGraphqlMockServer(xhrMock, {
+      operationsByTarget: { mc: operations },
+    });
+    selectProjectKeyFromUrl.mockReturnValue(
+      operations.FetchLoggedInUser.me.defaultProjectKey
+    );
+  });
+  it('should not render item', async () => {
+    const navbarMock = ApplicationNavbarMenuMock.build({
+      permissions: ['ManageOrders'],
+      actionRights: [{ group: 'orders', name: 'AddOrders' }],
+    });
+    const rendered = renderApp(null, {
+      featureFlags: {
+        [PROJECT_EXTENSIONS]: false,
+      },
+      DEV_ONLY__loadNavbarMenuConfig: () => Promise.resolve([navbarMock]),
+    });
+    // Get the nav container, to narrow down the search area
+    const container = await rendered.findByTestId('left-navigation');
+    const navbarRendered = within(container);
+
+    const applicationLocale = operations.FetchLoggedInUser.me.language;
+    const mainMenuLabel = navbarMock.labelAllLocales.find(
```

Comment:
I "trick" I did the other day to keep tests shorter was to return e.g. `return { mainMenuLabel, renderedNavBar, ...rendered}` etc from `render`. This encapsulates things a bit to keep tests lean. That would maybe help avoiding repetition here too?